Modules/_datetimemodule.c

```diff
@@ -4473,22 +4473,24 @@ time_richcompare(PyObject *self, PyObject *other, int op)
     }
     /* The hard case: both aware with different UTC offsets */
     else if (offset1 != Py_None && offset2 != Py_None) {
-        int offsecs1, offsecs2;
+        int64_t offusecs1, offusecs2, diff64;
```

Comment:
This might be a minor discussion point, but we could avoid using `int64_t` and the following big-integer calculations by doing a multi-step comparison, i.e. comparing the hours, then minutes, then seconds etc. 
The motivation for this is a bit of gut feeling, but I think that way will be slightly more efficient in terms of time and space.

Reply:
I actually implemented it like you suggest earlier, because I share your gut feeling. Using the existing code for checks up to seconds and adjusting only the microseconds check. Indeed with the advantage that it all fits in 32 bits.
 
What I encountered using this approach was the following: The combined microseconds values, in the time values and the offsets, can contribute up to a maximum of 1.999998 seconds to the diff in either way. That means that the simple intermediary check that is used in other multi step comparisons ` if (diff == 0)` can not be used. It should become `if (diff > -2 && diff < -2)`. I chose to go for the int64_t approach because I thought it to be less confusing, than the 2 seconds intermediary check, but I am perfectly willing to go with your suggestion, if that is the consensus.

Reply:
@blenq Does one of the tests already handle the use case of `13:59:59.9+02` vs. `14:00:00+02:00:00.9`?  If not we should add that as a test case.